### PR TITLE
Adds example with usb midi device

### PR DESCRIPTION
**Pull request by michalkalinowski-**
## Adds dev_midi example
Ports `midi_test` example from [tinyUSB repo](https://github.com/hathach/tinyusb/tree/master/examples/device/midi_test). I've cleaned up `tusb_config.h` a bit and fixed up code formatting to match other pico examples.

### Testing
Verified that the example builds and sends midi messages (screen shot of Pocket MIDI displaying incoming note on and note off messages):
Screenshot 2021-02-19 at 10 27 28

### CODE REVIEW

**Comment by kilograham** [comment]
can you please rebase this onto develop

**Comment by infovore** [comment]
@michalkalinowski- hah, I just did [almost the same thing](https://github.com/infovore/pico-example-midi) - should have checked the open PRs first!

**Comment by michalkalinowski-** [comment]
@kilograham sorry for a late reply, just rebased on develop.

**Comment by lurch** [comment]
@michalkalinowski- Can you click the "Edit" button at the top of this PR and change the base branch to `raspberrypi:develop` please? That should "clean up" the `Commits` tab in this PR :slightly_smiling_face:

**Comment by michalkalinowski-** [comment]
> @michalkalinowski- Can you click the "Edit" button at the top of this PR and change the base branch to `raspberrypi:develop` please? That should "clean up" the `Commits` tab in this PR 🙂

Hmm, I still see #30 among my commits, let me see if I can quickly fix that.

**Comment by michalkalinowski-** [comment]
All right, the diff looks ok now :)

File: README.md [hunk 1]
```diff
@@ -182,6 +182,7 @@ App|Description
 [dev_hid_composite](usb/device/dev_hid_composite) | Composite HID (mouse + keyboard) example from TinyUSB
 [dev_hid_generic_inout](usb/device/dev_hid_generic_inout) | Generic HID device example from TinyUSB
 [dev_lowlevel](usb/device/dev_lowlevel) | A USB Bulk loopback implemented with direct access to the USB hardware (no TinyUSB)
+[dev_midi](usb/device/dev_midi) | USB midi device example from TinyUSB
```
Comment (lurch) [inline README.md:185]: I wonder if it's worth mentioning here that this just sends dummy note data to the host PC? (i.e. it isn't (yet) a fully-fledged USB MIDI hardware interface) :man_shrugging:

Comment (michalkalinowski-) [reply README.md]: Good idea, I've updated the readme.

**Comment by kilograham** [comment]
Note, this will be subsumed by #99 where we will be using all TinyUSB examples directly

**Comment by raveslave** [comment]
now when we have this (usb host) support https://github.com/raspberrypi/pico-examples/blob/master/usb/host/host_cdc_msc_hid/main.c
anyone that would be up for adding usb midi host support as well?

**Pull request closed without merging**